src/platform/linux.c

```diff
@@ -592,8 +592,8 @@ void platform_get_user_directory(utf8 *outPath, const utf8 *subDirectory)
 	safe_strncpy(buffer, _userDataDirectoryPath, sizeof(buffer));
 	if (subDirectory != NULL && subDirectory[0] != 0) {
 		log_verbose("adding subDirectory '%s'", subDirectory);
-		strcat(buffer, subDirectory);
-		strcat(buffer, separator);
+		strncat(buffer, subDirectory, MAX_PATH);
```

Comment:
This is just as unsafe as the previous version it can still walk off the end of MAX_PATH. You've corrected this mistake on a strncat lower down.